test/unit/math/prim/mat/vectorize/foo_fun.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_FOO_FUN_HPP
```

Comment:
The foo test is in there both as an example of how to vectorize
a function and as a mock unit test of the functionality of
the feature.

As Rayleigh points out, exp() is not a good general candidate,
but log() would work, or a combo of log() and exp().  That's going to
drag this request out a fair bit as we need to

1.  vectorize the functions

2.  remove the old vectorizations if any

3.  write a general section of doc on vectorization (I can do this if
Rayleigh doesn't want to), which means we need to come up with a notation
for 

  promote_args<T>::type exp(T)

where T can be any of the vectorized types.  Right now, we just have the
two signatures done explicitly for the regular functions and have
the "reals" notation for the vectorized densities, which work very
differently.

- Bob



> On Mar 1, 2016, at 10:19 PM, Daniel Lee <notifications@github.com> wrote:
> 
> In test/unit/math/prim/mat/vectorize/foo_fun.hpp:
> 
> > @@ -0,0 +1,34 @@
> > +#ifndef TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_FOO_FUN_HPP
> 
> can we remove this function and just use exp as an example?
> 
> —
> Reply to this email directly or view it on GitHub.
>